examples/bridge-app/linux/include/Device.h

```diff
@@ -18,230 +18,59 @@
 
 #pragma once
 
-#include <app/util/attribute-storage.h>
+#include <lib/support/Span.h>
+#include <app-common/app-common/zap-generated/cluster-enums.h>
 
 #include <stdbool.h>
 #include <stdint.h>
 
-#include <functional>
-#include <vector>
-
+#include "Clusters.h"
+
+// Device types for dynamic endpoints: TODO Need a generated file from ZAP to define these!
+// (taken from chip-devices.xml)
+#define DEVICE_TYPE_BRIDGED_NODE 0x0013
+// (taken from lo-devices.xml)
+#define DEVICE_TYPE_LO_ON_OFF_LIGHT 0x0100
+// (taken from lo-devices.xml)
+#define DEVICE_TYPE_LO_ON_OFF_LIGHT_SWITCH 0x0103
+// (taken from chip-devices.xml)
+#define DEVICE_TYPE_ROOT_NODE 0x0016
+// (taken from chip-devices.xml)
+#define DEVICE_TYPE_BRIDGE 0x000e
+
+// Device Version for dynamic endpoints:
+#define DEVICE_VERSION_DEFAULT 1
+
+// This represents a single logical device occupying one endpoint. A composed device consists of multiple
+// Device objects that reference a tree.
 class Device
 {
 public:
-    static const int kDeviceNameSize = 32;
+    Device(chip::Span<chip::DataVersion> dataVersions, chip::Span<EmberAfCluster> clusters, chip::Span<ClusterImpl *> clusterImpl,
+           const chip::Span<const EmberAfDeviceType> & deviceTypeList, chip::EndpointId parentId);
+    ~Device() = default;
 
-    enum Changed_t
-    {
-        kChanged_Reachable = 1u << 0,
-        kChanged_Location  = 1u << 1,
-        kChanged_Name      = 1u << 2,
-        kChanged_Last      = kChanged_Name,
-    } Changed;
+    const chip::Span<chip::DataVersion> & versions() { return mDataVersions; }
+    const chip::Span<const EmberAfDeviceType> & deviceTypes() { return mDeviceTypeList; }
+    const EmberAfEndpointType * endpointType() { return &mEndpointType; }
+    const chip::Span<ClusterImpl *> & clusters() { return mClusterImpl; }
 
-    Device(const char * szDeviceName, std::string szLocation);
-    virtual ~Device() {}
+    void SetEndpointId(chip::EndpointId id);
+    chip::EndpointId GetEndpointId() { return mEndpointId; }
+    chip::EndpointId GetParentEndpointId() { return mParentEndpointId; }
 
-    bool IsReachable();
-    void SetReachable(bool aReachable);
-    void SetName(const char * szDeviceName);
-    void SetLocation(std::string szLocation);
-    inline void SetEndpointId(chip::EndpointId id) { mEndpointId = id; };
-    inline chip::EndpointId GetEndpointId() { return mEndpointId; };
-    inline void SetParentEndpointId(chip::EndpointId id) { mParentEndpointId = id; };
-    inline chip::EndpointId GetParentEndpointId() { return mParentEndpointId; };
-    inline char * GetName() { return mName; };
-    inline std::string GetLocation() { return mLocation; };
-    inline std::string GetZone() { return mZone; };
-    inline void SetZone(std::string zone) { mZone = zone; };
+    const char * GetName();
+    void SetName(const char * name);
```

Comment:
Should we have a comment regarding lifetimes for this?

or given that we are on a large platform, how about storing the name as a std::string and not worry about lifetimes of pointers.

Reply:
Just made it std::string. Can worry about lifetimes later if needed